src/kiragine/sharedtypes.zig

```diff
@@ -163,7 +108,7 @@ pub const Texture = struct {
 
         if (data == null) {
             gl.texturesDelete(1, @ptrCast([*]u32, &result.id));
-            return Error.FailedToLoadTexture;
+            return error.FailedToLoadTexture;
```

Comment:
Hmm, it makes sense but, zig compiler does not allow this code:
```zig
const t = engine.Texture.createFromPNG("ddd") catch |err| {
	switch (err) {
	    error.FailedToLoadTexture => {},
	}
};
```
```
zig version
0.6.0+9fe4c8923
```

```zig
error: expected type '.kiragine.sharedtypes.Texture', found 'void'
    t = engine.Texture.createFromPNG("ddd") catch |err| {
```

Reply:
Even though it makes sense, the reason why there is a huge global error set, it makes it easier to know what error might occur. Yes i can make a comment for every function that `may return this error` but that would be hard to know actually what error might can show up in functions if something is changed and documentation was not updated and may lead to confusion for other developers

Reply:
That is because the catch is an expression that needs to evaluate to the same type as `createFromPNG` or return.

For example, this will set `texture` to `standInTexture` if the texture could not be loaded:
```zig
const standInTexture: engine.Texture = getStandInTexture();

const texture = engine.Texture.createFromPNG("ddd") catch |err| {
    switch (err) {
        .FailedToLoadTexture => standInTexture,
    }
};
```

Or if continuing is not possible (instead of logging and returning it could instead panic or return an error itself):
```zig
const texture = engine.Texture.createFromPNG("ddd") catch |err| {
    switch (err) {
        .FailedToLoadTexture => {
            std.log.err("Log Something", .{});
            return;
        },
    }
};
```

Reply:
> Or if continuing is not possible (instead of logging and returning it could instead panic or return an error itself):
```zig
const texture = engine.Texture.createFromPNG("ddd") catch |err| {
    switch (err) {
        .FailedToLoadTexture => {
            std.log.err("Log Something", .{});
            return;
        },
    }
};
```
It still does not compile, how does it compiles on your compiler? Whats your zig version?
```zig
error: expected type '@typeInfo(@typeInfo(@TypeOf(.kiragine.sharedtypes.Texture.createFromPNG)).Fn.return_type.?).ErrorUnion.error_set', found '(enum literal)'
            .FailedToLoadTexture => {
            ^
./examples/textures.zig:43:9: note: referenced here
        switch (err) {
        ^
```

Reply:
It's been 3 days, it does not work with current version of zig. Closing the PR